src/Traits/DetectsChanges.php

```diff
@@ -133,8 +133,9 @@ public static function logChanges(Model $model): array
                     static::getModelAttributeJsonValue($model, $attribute)
                 );
             } else {
-                $changes[$attribute] = $model->getAttribute($attribute);
-
+                $changes[$attribute] = $model->hasCast($attribute, 'array') && is_array($model->attributes[$attribute])
```

Comment:
The `json` cast is the same. And does this happen with `collections`/`objects` too? 🤔